master/internal/rm/kubernetesrm/pods.go

```diff
@@ -102,9 +102,13 @@ type pods struct {
 	podInterfaces       map[string]typedV1.PodInterface
 	configMapInterfaces map[string]typedV1.ConfigMapInterface
 
+	// TODO(RM-236) make one cache and make this code more straightforward.
```

Comment:
The code is just really confusing. Summarize by resource pool and get agents look very similar but sometimes act differently. I think the code could probably benefit from a significant refactor that I don't want to do here.

Reply:
Makes sense. For what it's worth, since the cache is just a single data structure, I kinda like that they're separate ... but I can understand how it would benefit from a refactor either way.

Reply:
To me its more like we should have one way to summarize and get the cluster state. That cluster state summarization should be behind a cache. 

Then for get agents which needs the data in a certain way we should process the cluster state from the cache and return the needed response

For get resource pools we should process that same cluster state and return that response

We should decouple the cluster state from the apis that consume it